plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/RetryingJdbcClient.java

```diff
@@ -0,0 +1,527 @@
+        return retry(policy, () -> delegate.getSchemaNames(session));
+    }
+
+    @Override
+    public List<SchemaTableName> getTableNames(ConnectorSession session, Optional<String> schema)
+    {
+        return retry(policy, () -> delegate.getTableNames(session, schema));
+    }
+
+    @Override
+    public Optional<JdbcTableHandle> getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        return retry(policy, () -> delegate.getTableHandle(session, schemaTableName));
+    }
+
+    @Override
+    public JdbcTableHandle getTableHandle(ConnectorSession session, PreparedQuery preparedQuery)
+    {
+        return retry(policy, () -> delegate.getTableHandle(session, preparedQuery));
+    }
+
+    @Override
+    public JdbcProcedureHandle getProcedureHandle(ConnectorSession session, ProcedureQuery procedureQuery)
+    {
+        return retry(policy, () -> delegate.getProcedureHandle(session, procedureQuery));
+    }
+
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
+    {
+        return retry(policy, () -> delegate.getColumns(session, tableHandle));
+    }
+
+    @Override
+    public Iterator<RelationColumnsMetadata> getAllTableColumns(ConnectorSession session, Optional<String> schema)
+    {
+        return retry(policy, () -> delegate.getAllTableColumns(session, schema));
+    }
+
+    @Override
+    public List<RelationCommentMetadata> getAllTableComments(ConnectorSession session, Optional<String> schema)
+    {
+        return retry(policy, () -> delegate.getAllTableComments(session, schema));
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        // no retrying as it could be not idempotent operation (connection could be not reusable after the first failure)
+        return delegate.toColumnMapping(session, connection, typeHandle);
+    }
+
+    @Override
+    public List<ColumnMapping> toColumnMappings(ConnectorSession session, List<JdbcTypeHandle> typeHandles)
+    {
+        // there should be no remote database interaction
+        return delegate.toColumnMappings(session, typeHandles);
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        // there should be no remote database interaction
+        return delegate.toWriteMapping(session, type);
+    }
+
+    @Override
+    public Optional<Type> getSupportedType(ConnectorSession session, Type type)
+    {
+        // there should be no remote database interaction
+        return delegate.getSupportedType(session, type);
+    }
+
+    @Override
+    public boolean supportsAggregationPushdown(ConnectorSession session, JdbcTableHandle table, List<AggregateFunction> aggregates, Map<String, ColumnHandle> assignments, List<List<ColumnHandle>> groupingSets)
+    {
+        // there should be no remote database interaction
```

Comment:
I think i disagree with the choice that since there are no remote interactions today so there should be no retries.

IMO a `RetryingJdbcClient` should retry everything except things we know cannot be retried (e.g. `finishCreateTable` as a whole might not be retriable depending on the exception)

Reply:
I don't have very strong opinion in this case. However I wanted to avoid to overkill with retries for simple things. Retrying is also some amount of code to execute so I didn't want to do it for simple things.

Reply:
Ok, in that case I'll just review if the cases where we've skipped it make sense or not.